google-cloud-core/src/main/java/com/google/cloud/Policy.java

```diff
@@ -157,66 +174,147 @@ protected Builder(Policy policy) {
      * Replaces the builder's map of bindings with the given map of bindings.
      *
      * @throws NullPointerException if the given map is null or contains any null keys or values
-     * @throws IllegalArgumentException if any identities in the given map are null
+     * @throws IllegalArgumentException if any identities in the given map are null or if policy
+     *     version is equal to 3 or has conditional bindings.
      */
     public final Builder setBindings(Map<Role, Set<Identity>> bindings) {
       checkNotNull(bindings, "The provided map of bindings cannot be null.");
+      checkArgument(
```

Comment:
These should all throw `IllegalStateException` instead (using `checkState`), since the problem isn't with the arguments to the methods but about whether the method can be called at all given its state.